sql-plugin/src/main/scala/org/apache/spark/sql/rapids/metrics/source/MockTaskContext.scala

```diff
@@ -76,5 +76,7 @@ class MockTaskContext(taskAttemptId: Long, partitionId: Int) extends TaskContext
   override private[spark] def fetchFailed = None
 
   override private[spark] def getLocalProperties = new Properties()
+
+  def cpus(): Int = 2
```

Comment:
seems out of place, why this change?

Reply:
because it's required for 3.3 SPARK-36173 
 https://github.com/apache/spark/commit/f6e6d1157ac988d7c5809fcb08b577631bdea8eb